theano/sandbox/cuda/cula.py

```diff
@@ -0,0 +1,103 @@
+import theano
+from theano.sandbox.cuda.type import CudaNdarrayType
+from theano.sandbox.cuda import GpuOp, CudaNdarray
+from theano.sandbox.cuda.basic_ops import (as_cuda_ndarray_variable,
+                                           gpu_contiguous)
+from theano.tensor import as_tensor_variable
+from scikits.cuda import cula
+try:
+    from scikits.cuda import cula
+    scikits_cuda_available = True
+except ImportError:
+    scikits_cuda_available = False
+
+import numpy
+
+class GpuSolve(GpuOp):
+    """
+    CULA GPU solver OP.
+
+    """
+    def __init__(self, trans='N'):
+        self.trans = trans
+        super(GpuSolve, self).__init__()
+
+    def __eq__(self, other):
+        return (type(other) == type(self))
+
+    def __hash__(self):
+        return hash(type(self))
+
+    def output_type(self, inp):
+        return CudaNdarrayType(broadcastable=[False] * inp.type.ndim)
+
+    def make_node(self, inp1, inp2):
+        inp1 = gpu_contiguous(as_cuda_ndarray_variable(inp1))
+        inp2 = gpu_contiguous(as_cuda_ndarray_variable(inp2))
```

Comment:
Actually, that may not be such a good idea to do that, since we have to make a new copy anyway in the `thunk`. We should have only one copying operation in the `thunk` instead.